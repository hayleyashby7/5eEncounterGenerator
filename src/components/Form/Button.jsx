/* eslint-disable react/button-has-type */
/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';

function Button({ label, type, ...rest }) {
    return (
        <button
            onClick={rest.onClick}
            type={type}
            className="rounded-lg border-2 border-solid border-orange-100 bg-red-950 p-2 font-['Alegreya_Sans'] text-orange-100 hover:bg-red-900 hover:text-orange-200 focus:outline-none focus:ring-2 focus:ring-red-900/50"
            {...rest}
        >
            {label}
        </button>
    );
}

export default Button;

Button.propTypes = {
    label: PropTypes.string,
    type: PropTypes.string,
    rest: PropTypes.object,
};
Button.defaultProps = {
    label: '',
    type: 'button',
    rest: {},
};
