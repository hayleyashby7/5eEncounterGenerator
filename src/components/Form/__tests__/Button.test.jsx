import { describe, it, expect, vi, beforeEach } from 'vitest';
import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '../Button';

describe('Button component', () => {
    beforeEach(() => {
        vi.clearAllMocks();
    });

    it('renders the button', () => {
        // Arrange
        const { container } = render(<Button />);

        // Assert
        expect(container).toMatchSnapshot();
    });

    it('should render a button with the label passed in', () => {
        // Arrange
        const label = 'Test label';

        // Act
        render(<Button label={label} />);

        // Assert
        expect(screen.getByRole('button')).toHaveTextContent(label);
    });

    it('should render a button with the correct type', () => {
        // Arrange
        const type = 'submit';

        // Act
        render(<Button type={type} />);

        // Assert
        expect(screen.getByRole('button')).toHaveAttribute('type', type);
    });

    it('should call the onClick function when the button is clicked', async () => {
        // Arrange
        const mockOnClick = vi.fn();
        const type = 'button';
        const label = 'Test label';
        const user = userEvent.setup();

        render(<Button type={type} label={label} onClick={mockOnClick} />);

        // Act
        await user.click(screen.getByRole('button'));

        // Assert
        expect(mockOnClick).toBeCalled();
        expect(mockOnClick).toBeCalledTimes(1);
    });
});
