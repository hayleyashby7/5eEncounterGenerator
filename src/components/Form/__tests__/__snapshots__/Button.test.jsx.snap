// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button component > renders the button 1`] = `
<div>
  <button
    class="rounded-lg border-2 border-solid border-orange-100 bg-red-950 p-2 text-orange-100 hover:bg-red-900 hover:text-orange-200 focus:outline-none focus:ring-2 focus:ring-red-900/50"
    rest="[object Object]"
    type="button"
  />
</div>
`;
